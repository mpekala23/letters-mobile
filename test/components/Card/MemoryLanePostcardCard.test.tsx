import React from 'react';
import { MemoryLanePostcardCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

const setup = (propOverrides = {}) => {
  const props = {
    date: new Date('2019-07-12T15:51:41.000Z'),
    imageUri: 'test',
    color: 'green',
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<MemoryLanePostcardCard {...props} />),
    props,
  };
};

describe('MemoryLanePhotoCard component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on a press', () => {
    const { props, getByTestId } = setup();
    fireEvent.press(getByTestId('memoryLanePostcardImage'));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should display date', () => {
    const { getByText } = setup();
    expect(getByText('July 12')).toBeDefined();
  });

  it('should display memory lane image', () => {
    const { getByTestId } = setup();
    expect(getByTestId('memoryLanePostcardImage').props.source.uri).toBe(
      'test'
    );
  });
});
