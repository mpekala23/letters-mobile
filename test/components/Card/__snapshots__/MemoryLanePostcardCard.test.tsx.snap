// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryLanePhotoCard component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "borderRadius": 6,
          "height": 210,
          "padding": 0,
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
    testID="memoryLanePostcardCard"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderRadius": 6,
          "height": "100%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "test",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Object {
              "borderRadius": 6,
            },
          ]
        }
        testID="memoryLanePostcardImage"
      />
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "backgroundColor": "rgba(0,0,0, 0.30)",
              "borderRadius": 6,
              "flex": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "marginBottom": 8,
                "marginTop": "auto",
              },
              Object {
                "fontFamily": "Poppins-Bold",
              },
            ]
          }
        >
          July 12
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
